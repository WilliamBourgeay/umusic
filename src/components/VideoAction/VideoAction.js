// @flow

import * as React from 'react';
import IconButton from '@material-ui/core/IconButton';

type Props = {
  hide: boolean,
  className?: string,
  children: React.Node,
  ariaLabel?: string,
  onClick: Function,
};

const VideoAction = ({ hide, onClick, className, children, ariaLabel }: Props) => {
  if (hide) return null;

  return (
    <IconButton color="primary" onClick={onClick} className={className} aria-label={ariaLabel}>
      {children}
    </IconButton>
  );
};

VideoAction.defaultProps = {
  className: '',
  ariaLabel: '',
};

export default VideoAction;
