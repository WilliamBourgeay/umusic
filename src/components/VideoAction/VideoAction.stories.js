import React from 'react';
import { storiesOf } from '@storybook/react';
import Pause from '@material-ui/icons/Pause';

import VideoAction from './VideoAction';

const stories = storiesOf('VideoAction', module);

const createVideoAction = () => (
  <VideoAction onClick={() => {}} hide={false}>
    <Pause fontSize="large" />
  </VideoAction>
);

stories.add('default', () => createVideoAction());
