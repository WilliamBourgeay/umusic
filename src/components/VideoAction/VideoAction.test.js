import * as React from 'react';
import SearchIcon from '@material-ui/icons/Search';
import renderer from 'react-test-renderer';

import VideoAction from './VideoAction';

describe('VideoAction', () => {
  const onClick = () => {};

  it('should render a VideoAction ', () => {
    const tree = renderer.create(
      <VideoAction hide={false} onClick={onClick}>
        <SearchIcon />
      </VideoAction>,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
