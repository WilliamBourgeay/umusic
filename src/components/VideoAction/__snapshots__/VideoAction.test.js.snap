// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoAction should render a VideoAction  1`] = `
<button
  aria-label=""
  className="MuiButtonBase-root-7 MuiIconButton-root-1 MuiIconButton-colorPrimary-3"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button"
>
  <span
    className="MuiIconButton-label-6"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-10"
      color={undefined}
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
      />
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
    </svg>
  </span>
  <span
    className="MuiTouchRipple-root-19"
  />
</button>
`;
